docs/swarm-api.md

```diff
@@ -24,71 +27,157 @@ POST "/images/create" : "docker import" flow not implement
+        <th>Endpoint</th>
+        <th>Differences</th>
+    </tr>
+    <tr>
+        <td>
+            <code>GET "/containers/{name:.*}/json"</code>
+        </td>
+        <td>
+            New field <code>Node</code> added:<br />
+<pre>
+"Node": {
+    "Id": "ODAI:IC6Q:MSBL:TPB5:HIEE:6IKC:VCAM:QRNH:PRGX:ERZT:OK46:PMFX",
+    "Ip": "0.0.0.0",
+    "Addr": "http://0.0.0.0:4243",
+    "Name": "vagrant-ubuntu-saucy-64"
+}
+</pre>
+        </td>
+    </tr>
+    <tr>
+        <td>
+            <code>GET "/containers/{name:.*}/json"</code>
+        </td>
+        <td>
+            <code>HostIP</code> replaced by the the actual Node's IP if <code>HostIP</code> is <code>0.0.0.0</code>
+        </td>
+    </tr>
+    <tr>
+        <td>
+            <code>GET "/containers/json"</code>
+        </td>
+        <td>
+            Node's name prepended to the container name.
+        </td>
+    </tr>
+    <tr>
+        <td>
+            <code>GET "/containers/json"</code>
+        </td>
+        <td>
+            <code>HostIP</code> replaced by the the actual Node's IP if <code>HostIP</code> is <code>0.0.0.0</code>
+        </td>
+    </tr>
+    <tr>
+        <td>
+            <code>GET "/containers/json"</code>
+        </td>
+        <td>
+            Containers started from the <code>swarm</code> official image are hidden by default, use <code>all=1</code> to display them.
+        </td>
+    </tr>
+    <tr>
+        <td>
+            <code>GET "/images/json"</code>
+        </td>
+        <td>
+            Use <code>--filter node=&lt;Node name&gt;</code> to show images of the specific node.
+        </td>
+    </tr>
+    <tr>
+        <td>
+            <code>POST "/containers/create"</code>
+        </td>
+        <td>
+            <code>CpuShares</code> in <code>HostConfig</code> sets the number of CPU cores allocated to the container.
+        </td>
+    </tr>
+</table>
+
+## Registry Authentication
+
+During container create calls, the swarm API will optionally accept an `X-Registry-Auth` header.
+If provided, this header is passed down to the engine if the image must be pulled
+to complete the create operation.
+
+The following two examples demonstrate how to utilize this using the existing Docker CLI
+
+### Authenticate using registry tokens
+
+> **Note:** This example requires Docker Engine 1.10 with auth token support.
+> For older Engine versions, refer to [authenticate using username and
+> password](#authenticate-using-username-and-password)
+
+This example uses the [`jq` command-line utility](https://stedolan.github.io/jq/).
+To run this example, install `jq` using your package manager (`apt-get install jq` or `yum install jq`).
+
+```bash
+REPO=yourrepo/yourimage
+REPO_USER=yourusername
+read -s PASSWORD
+AUTH_URL=https://auth.docker.io/token
+
+# obtain a JSON token, and extract the "token" value using 'jq'
+TOKEN=$(curl -s -u "${REPO_USER}:${PASSWORD}" "${AUTH_URL}?scope=repository:${REPO}:pull&service=registry.docker.io" | jq -r ".token")
```

Comment:
Note that we could replace `jq` with (e.g.) `sed` to prevent people from having to install `jq` (`sed` is more commonly installed). This seems to work (both on Linux and OS X);

```bash
curl -s -u "${REPO_USER}:${PASSWORD}" "${AUTH_URL}?scope=repository:${REPO}:pull&service=registry.docker.io" | sed -n 's|.*"token":"\([^"]*\)".*|\1|p'
```

Or, to test;

```bash
echo '{"token":"foobar"}' | sed -n 's|.*"token":"\([^"]*\)".*|\1|p'
```